packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1163,17 +1185,205 @@ class _InactiveElements {
+///   @override
+///   Widget build(BuildContext context) {
+///     // here, Scaffold.of(context) returns null
+///     return new Scaffold(
+///       appBar: new AppBar(title: new Text('Demo')),
+///       body: new Builder(
+///         builder: (BuildContext context) {
+///           return new FlatButton(
+///             child: new Text('BUTTON'),
+///             onPressed: () {
+///               // here, Scaffold.of(context) returns the locally created Scaffold
+///               Scaffold.of(context).showSnackBar(new SnackBar(
+///                 content: new Text('Hello.')
+///               ));
+///             }
+///           );
+///         }
+///       )
+///     );
+///   }
+/// ```
+///
+/// The [BuildContext] for a particular widget can change location over time as
+/// the widget is moved around the tree. Because of this, values returned from
+/// the methods on this class should not be cached beyond the execution of a
+/// single synchronous function.
 abstract class BuildContext {
+  /// The current configuration of the widget to which this [BuildContext] belongs.
   Widget get widget;
+
+  /// The current [RenderObject] for the widget. If the widget is a
+  /// [RenderObjectWidget], this is the render object that the widget created
+  /// for itself. Otherwise, it is the render object of the first descendant
+  /// [RenderObjectWidget].
+  ///
+  /// This method will only return a valid result after the build phase is
+  /// complete. It is therefore not valid to call this from the [build] function
+  /// itself. It should only be called from interaction event handlers (e.g.
+  /// gesture callbacks) or layout or paint callbacks.
+  ///
+  /// If the render object is a [RenderBox], which is the common case, then the
+  /// size of the render object can be obtained from the [RenderBox.size]
+  /// getter. This is only valid after the layout phase, and should therefore
+  /// only be examined from paint callbacks or interaction event handlers (e.g.
+  /// gesture callbacks).
+  ///
+  /// Calling this method is theoretically relatively expensive (O(N) in the
+  /// depth of the tree), but in practice is usually cheap because the tree
+  /// usually has many render objects and therefore the distance to the nearest
+  /// render object is usually short.
   RenderObject findRenderObject();
+
+  /// Obtains the nearest widget of the given type, which must be the type of a
+  /// concrete [InheritedWidget] subclass, and registers this build context with
+  /// that widget such that when that widget changes (or a new widget of that
+  /// type is introduced, or the widget goes away), this build context is
+  /// rebuilt so that it can obtain new values from that widget.
+  ///
+  /// This is typically called implicitly from `of()` static methods, e.g.
+  /// [Theme.of].
+  ///
+  /// This is safe to call from build methods, layout and paint callbacks,
+  /// interaction event handlers (e.g. gesture callbacks), and so forth.
```

Comment:
Is this the place to explain that the method shouldn't be called from initState()  or a Widget constructor?